core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -93,7 +93,7 @@ class BrokerLifecycleManagerImpl(val brokerMetadataListener: BrokerMetadataListe
   private var schedulerTask: Option[ScheduledFuture[_]] = None
 
   // Broker states - Current and Target/Next
-  private var currentState: jmetadata.BrokerState = _
+  private var currentState: jmetadata.BrokerState = jmetadata.BrokerState.REGISTERING
```

Comment:
Thanks @rondagostino! I think this change would fail a couple of the tests in BrokerLifecycleManagerTest.scala considering they expect the initial state (of `NOT_RUNNING`) in some registration failure scenarios

Reply:
@aloknnikhil All tests pass, actually.  I learned today that recovery from an unclean shutdown happens synchronously when LogManager is instantiated -- the instantiation doesn't return until any recovery is complete.  So the heartbeat doesn't need to worry about that state at all, and it can jump right into registration.

Reply:
That's great. Definitely keeps it clean.

Reply:
I think we should move the stuff about broker registration out of this class, and just handle heartbeats here.  That avoids the really awkward blocking in the `start` function.  We should also try to avoid keeping the current BrokerState here (we already store that in Kip500Broker, let's not duplicate things)

Reply:
**Registration**:
- I think moving it out should be OK. But considering the `BrokerLifecycleManager` owns the lifecycle and any state transitions, I figured registering the broker would be the first step in starting the lifecycle manager.

**BrokerState**:
- The Kip500Broker function references the state maintained by the `BrokerLifecycleManager`. I think it makes sense to use that as the source of truth and leave it in the lifecycle manager considering it's the LifecycleManager that either approves or denies the state  transitions based on the response from the controller.